pkg/operator/storageclasscontroller/vmware.go

```diff
@@ -205,6 +328,7 @@ func (v *storagePolicyAPI) createStorageProfile(ctx context.Context) error {
 
 	var policySpec types.PbmCapabilityProfileCreateSpec
 	policySpec.Name = v.policyName
+	policySpec.Category = "REQUIREMENT"
```

Comment:
This change is surprising. Was the storage profile created in OCP 4.12 correct without this field?

Reply:
I had to make this change mostly to keep `vcsim` client happy. It appears that - with real cluster, this field is automatically set and it works as expected. But with `vcsim` client - when we don't set this field, it caches the object "as it is" and then when we query for this profile - it can't find it. Because it expectes `Category == REQUIREMENT`. 

Not sure how to do get out of this. I could not modify cached profiles because `vcsim` package stores that as a private unexported field and  I can't do modify it...

Reply:
I also tested, it doesn't break real API calls - so I thought it is fine.

Reply:
ack